### PR TITLE
Custom offset functionality for Popup Menu Button

### PR DESCRIPTION
**Pull request by jslavitz**
Adds love to #20861

### CODE REVIEW

File: packages/flutter/lib/src/material/popup_menu.dart [hunk 1]
```diff
@@ -862,6 +863,9 @@ class PopupMenuButton<T> extends StatefulWidget {
   /// If provided, the icon used for this button.
   final Icon icon;
 
+  /// The offset applied to the Popup Menu Button
+  final Offset offset;
```
Comment (jonahwilliams) [inline packages/flutter/lib/src/material/popup_menu.dart]: Describe what this offset does and document that the default value is zero.

File: packages/flutter/lib/src/material/popup_menu.dart [hunk 2]
```diff
@@ -818,6 +818,7 @@ class PopupMenuButton<T> extends StatefulWidget {
     this.padding = const EdgeInsets.all(8.0),
     this.child,
     this.icon,
+    this.offset = Offset.zero,
   }) : assert(itemBuilder != null),
```
Comment (jonahwilliams) [inline packages/flutter/lib/src/material/popup_menu.dart]: assert(offset != null)

File: packages/flutter/test/material/popup_menu_test.dart [hunk 3]
```diff
@@ -430,6 +430,43 @@ void main() {
...
+          child: new Material(
+            child: popupMenuButton,
+          ),
+        )
+    ));
+
+    await tester.tap(find.byType(IconButton));
+    await tester.pumpAndSettle();
+    
+    expect(tester.getTopLeft(find.byWidgetPredicate((Widget w) => '${w.runtimeType}' == '_PopupMenu<int>')), const Offset(8.0, 40.0));
```
Comment (jonahwilliams) [inline packages/flutter/test/material/popup_menu_test.dart]: Do we know what the 8 is from? Would add a comment if possible.

Comment (jslavitz) [reply packages/flutter/test/material/popup_menu_test.dart]: I'm not sure... What it looks to me actually, is for this specific popup button we can actually only modify if it is positioned on the left or right side of the screen, and then it's vertical position on the screen.

File: packages/flutter/lib/src/material/popup_menu.dart (hunk 1 above)
Comment (jonahwilliams) [reply packages/flutter/lib/src/material/popup_menu.dart]: It's the offset applied to the popupMenu and not the button right?

```
The offset applied to the popup menu, relative to the button that opens it.

Defaults to [Offset.zero].
```

File: packages/flutter/lib/src/material/popup_menu.dart [hunk 4]
```diff
@@ -872,9 +880,9 @@ class _PopupMenuButtonState<T> extends State<PopupMenuButton<T>> {
   void showButtonMenu() {
     final RenderBox button = context.findRenderObject();
     final RenderBox overlay = Overlay.of(context).context.findRenderObject();
-    final RelativeRect position = RelativeRect.fromRect(
-      Rect.fromPoints(
-        button.localToGlobal(Offset.zero, ancestor: overlay),
+    final RelativeRect position = new RelativeRect.fromRect(
```
Comment (jonahwilliams) [inline packages/flutter/lib/src/material/popup_menu.dart]: Nit: remove `new`

File: packages/flutter/test/material/popup_menu_test.dart [hunk 5]
```diff
@@ -430,6 +430,44 @@ void main() {
...
+          ];
+        },
+      );
+
+    await tester.pumpWidget(new MaterialApp(
+        home: Scaffold(
+          body: Center(
+            child:  new Material(
+              child: popupMenuButton,
+          ),
```
Comment (jonahwilliams) [inline packages/flutter/test/material/popup_menu_test.dart]: Nit: I think these should be indented one more level

**Review by jonahwilliams — APPROVED** [review]
LGTM to land on green

**Pull request merged**